### PR TITLE
Documented affinity feature

### PR DESCRIPTION
**Pull request by stuclem**
Fixes https://github.com/vmware/vic-product/issues/1494. There is also some reorg towards https://github.com/vmware/vic-product/issues/1612.

Commit https://github.com/vmware/vic-product/commit/ec3e7fa620d74ef8b63508ed1908812425f3e3e2 includes the updates that relate specifically to affinity rules. @zjs can you please review? https://github.com/vmware/vic-product/pull/1722/commits/38792d3d0c0f58aca6f3d587863403ae168334a3 is the content reorg, which you do not need to review. Thanks!

I updated the topic on deleting VCHs to mention that the VM group is deleted. I couldn't find any changes in the behaviours of `vic-machine inspect`, `inspect config` or VCH Admin relating to affinity groups.

Also, do we need to update the topic on [manually configuring ops user](https://vmware.github.io/vic-product/assets/files/html/1.4/vic_vsphere_admin/ops_user_manual.html) to add `Host.Inventory.EditCluster`? Which role should we add that to?

### CODE REVIEW

File: docs/user_doc/vic_vsphere_admin/remove_vch.md [hunk 1]
```diff
@@ -13,15 +13,17 @@ The `vic-machine delete` includes one option in addition to the [Common `vic-mac
...
 
-     Use upper-case letters and colon delimitation in the thumbprint. Do not use space delimitation.
+**NOTES**:
 
 When you delete a VCH that uses TLS authentication with trusted Certificate Authority (CA) certificates, `vic-machine delete` does not delete the certificates or the certificate folder, even if you specify the `--force` option. Because `vic-machine delete` does not delete the certificates, you can delete VCHs and create new ones that reuse the same certificates. This is useful if you have already distributed the client certificates for VCHs that you need to recreate.
 
 The `vic-machine delete` command does not modify the firewall on ESXi hosts. If you do not need to deploy or run further VCHs on the ESXi host or cluster after you have deleted VCHs, run `vic-machine update firewall --deny` to close port 2377 on the host or hosts. 
 
-## Example ##
+If you deployed the VCH with `vic-machine create --affinity-vm-group option`, `vic-machine delete` removes the VM affinity group that was created during deployment.
```
Comment (zjs) [inline docs/user_doc/vic_vsphere_admin/remove_vch.md]: > with `vic-machine create --affinity-vm-group option`

should probably be either

> with the `vic-machine create --affinity-vm-group` option

or

> with `vic-machine create --affinity-vm-group`

File: docs/user_doc/vic_vsphere_admin/vch_affinity_group.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+When you deploy a virtual container host, you can optionally instruct vSphere Integrated Containers to automatically create a DRS VM group in vSphere for the VCH endpoint VM and its container VMs. If you use this option, you can  use the resulting VM group in DRS VM-Host affinity rules, to restrict the set of hosts on which the VCH endpoint VM and its container VMs can run.
+
+You might want to restrict the set of hosts on which the VCH and container VMs run for the following reasons:
+
+- Software licensing, for example if your organization is billed based on the number of physical hosts, sockets, or cores that run a particular piece of software.
+- Compliance with internal policies.
+- Latency-sensitivity, for workloads that run in an environment with stretched clusters.
+
+You can address each of these use cases by using DRS VM-host affinity groups.
```
Comment (zjs) [inline docs/user_doc/vic_vsphere_admin/vch_affinity_group.md]: > by using DRS VM-host affinity groups.

Nit:

> by using DRS VM-host affinity rules.

File: docs/user_doc/vic_vsphere_admin/vch_affinity_group.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+vSphere allows you to express VM-Host affinity rules either as a requirement (*must*/*must not* rules) or a preference (*should*/*should not* rules).
+
+  - If you define *must* rules, DRS does not allow the VMs to run on other hosts, even in extreme circumstances. For example, vSphere HA does not perform failovers to hosts that are not in the DRS host group. 
+  - If you define *should* rules, violations produce a log event and are reported as faults on the **Configure** > **vSphere DRS** view for the cluster.
+
+To set VM-Host affinity rules on a VCH, you perform the following steps:
+
+- In vSphere, create a DRS host group that includes the set of hosts to which to limit VCH and container VM workloads.
+- Deploy a VCH with the `vic-machine create --affinity-vm-group` option, which automatically creates a DRS VM group in vSphere for the VCH and its container VMs.
+- In vSphere, create a VM-Host affinity rule that includes the VM group and the host group. This ensures that the VCH and container VMs in the VM group only run on the hosts that you specified in the host group.
```
Comment (zjs) [inline docs/user_doc/vic_vsphere_admin/vch_affinity_group.md]: > the VCH and container VMs

nit: previous sections say "VCH endpoint VM and its container VMs", perhaps this should be consistent?

File: docs/user_doc/vic_vsphere_admin/vch_affinity_group.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+**IMPORTANT**: Because you define VM-host affinity rules on clusters, all of the hosts in a DRS host group must be reside in the same cluster.
```
Comment (zjs) [inline docs/user_doc/vic_vsphere_admin/vch_affinity_group.md]: > must be reside

should probably be 

> must reside

File: docs/user_doc/vic_vsphere_admin/remove_vch.md (hunk 1 above)
Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/remove_vch.md]: Fixed

File: docs/user_doc/vic_vsphere_admin/vch_affinity_group.md (hunk 2 above)
Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/vch_affinity_group.md]: Fixed

File: docs/user_doc/vic_vsphere_admin/vch_affinity_group.md (hunk 3 above)
Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/vch_affinity_group.md]: Fixed

File: docs/user_doc/vic_vsphere_admin/vch_affinity_group.md (hunk 4 above)
Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/vch_affinity_group.md]: Fixed

**Pull request merged**